src/cc/libbpf.c

```diff
@@ -173,12 +173,22 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
     }
   }
 
-  if (ret < 0 && !log_buf) {
-    // caller did not specify log_buf but failure should be printed,
-    // so call recursively and print the result to stderr
-    bpf_prog_load(prog_type, insns, prog_len, license, kern_version,
-        bpf_log_buf, LOG_BUF_SIZE);
-    fprintf(stderr, "bpf: %s\n%s\n", strerror(errno), bpf_log_buf);
+  if (ret < 0) {
+    const char *str_error = strerror(errno);
```

Comment:
Sure, but that would be outside scope of this bugfix. For now I would adopt the existing convention of `attach_raw_socket`.